### ISSUE TITLE
In App Purchase does not work after 7.4.11

### ISSUE THREAD

**Issue by JFClaeys** [renpy/renpy#3801]
I had a similar problem with issue https://github.com/renpy/renpy/issues/3095 which is now closed
After updated to 8.0.1, build does not return string of price while calling iap,get_price , and of course iap.get_purchase fails.
.android.json holds config that I recognized as to be valid
Contrary to previous case, though, the store name is displayed.
I tried to call iap.init right after label start ,in script.rpy, but that made no difference at all

Nightlies have been tested, as well as 7.5.1
This was tested on Google Play , as it's the only store we are using

In the example below, please find a test case, with , at the root,  .android.json .
The behaviour of the test is to try to collect the price of a few items as string.  From experience, if price can be fetch, then the  purchase can be done.
The items are real and form a real store.  Also, if running  7.4.11, they can be purchased

Note that key and salt have been modified with bogus values
[CompletBASEFolder.zip](https://github.com/renpy/renpy/files/9174518/CompletBASEFolder.zip)

**Comment by JFClaeys** [comment]
Thanks :)
Change is done ands internal test uploaded on the store
As soon as the uploaded version is tested, I will let you know the result

**Comment by JFClaeys** [comment]
I just tested and unfortunately, no change: still cannot see the prices.
When you mentionned the "Google Play" thing, you meant while debugging, using the "keys" I sent you?  These were bogus, of course.
I'm not sure it's a google play related problem, given that the machine using 7.4.11 is able to see the prices and as well to purchase.  Therefor, it has to be related, somehow, to a change somewhere in the code since then. At least from my point of view..

**Comment by renpytom** [comment]
Did you try building the actual game with the new Ren'Py, or just your test game. IIRC, there are restrictions on package name and account before you can see in-app purchases, so I don't think the test game you sent me would ever be able to see them. (The keys aren't used anymore.)

**Comment by renpytom** [comment]
I'm pretty sure this is working correctly. I set up a test app in Google Play, and was able to perform an in-app purchase. 

Can you post a few things: 

* A screenshot of the in-app products page from the Google Play developer console.
* A screenshot of the dashboard page from the Google Play developer console.
* The calls to iap.register from your game.
* The android.json file for your game (you can remove any key material). 

And finally, please add in your game:

```renpy
 $ store_name = iap.get_store_name()
"store_name = [store_name]
```

So we can see what Ren'Py things the store name is. I'd like to make sure everything matches up.

**Comment by JFClaeys** [comment]
Thanks for your time and involvement, highly appreciated

To answer your previous question : the real game was built using ver 8.0.1, on computer A, which was freshly upgraded from 7.4.11.  This is how we found out iap functions are not working (at least for us).  The same code was built afterward using computer B, which was still in version 7.4.11 and iap functions are working fine, using same keys and code.
The test game was written using the same keys as the actual game and was created to test https://github.com/renpy/renpy/issues/3095 to at least verify that iap.get_store_name  and  iap.get_price were functionning.
Test game is built on computer C which was initially on 7.4.11, then 8.0.1, then nightlies and finally on 7.5.1.  Last attempt was done on 7.5.1

Your next question :

```
init:
   define achat_cle_1 = iap.register(
     product = "achat_cle_1",
     identifier = "com.chimeres.achat.1cle",
     google = "com.chimeres.achat.1cle",
   )
 
   define achat_cle_5 = iap.register(
     product = "achat_cle_5",
     identifier = "com.chimeres.achat.5cle",
     google = "com.chimeres.achat.5cle",
   )
 
   define achat_fin_eric = iap.register(
     product = "achat_fin_eric",
     identifier = "com.chimeres.achat.fin.eric",
     google = "com.chimeres.achat.fin.eric",
   )
 
   define achat_fin_gabriel = iap.register(
     product = "achat_fin_gabriel",
     identifier = "com.chimeres.achat.fin.gabriel",
     google = "com.chimeres.achat.fin.gabriel",
   )
 
   define achat_fin_majid = iap.register(
     product = "achat_fin_majid",
     identifier = "com.chimeres.achat.fin.majid",
     google = "com.chimeres.achat.fin.majid",
   )
```

then
```
    if renpy.variant("android"):
       $ StoreName = iap.get_store_name()
       $ eric_price = iap.get_price("achat_fin_eric")
       $ gabriel_price = iap.get_price("achat_fin_gabriel")
       $ majhid_price = iap.get_price("achat_fin_majid")
    python:
        if renpy.windows:
           IsWindows = "True"
        else:
           IsWindows = "False"
```
followed by 

```
    "Store name  : [StoreName]      Version       : [renpy.version_string]"
    "Eric price  : [eric_price]     Gabriel price : [gabriel_price]         Majhid price : [majhid_price]"
    "Is Windows  : [IsWindows]      Is Android    : [renpy.android]"
```
the result, along with  the console screen shots you asked are included in the zip file attached.
Also, you will find the android.json and screen shots of what testor sees while running.  
Basically, it knows it is on play store, but all price inquires are returning None

Tonight, Ill build and load testor on computer A, with 7.4.11, as a matter of comparaison
[Screenies.zip](https://github.com/renpy/renpy/files/9180726/Screenies.zip)

**Comment by JFClaeys** [comment]
The product page is the one for Piegee, and the Testor was trying to access the iap of Piegee

Sorry for the delay, but your answer triggered a questioning/experimentation on my side.
The Google key seems to allow simply to access the Play Store, but not specifically the one of the app, as I though/experimented/deduced before.
It seems that there is another ID that does that now.  Cannot tell what it can be...
You see, in 7.4.11, my testor was able to access the store's products for Piegee : both apps were using the same Google Key.
But somehow, now, it can only see its own store (7.5.1 and more + fix).

Initially, there were no products in Testor's store.  
Only after adding them that the app is now able to make iap works.  So with your fix, it seems to be fine.
I will experiment a bit more before closing the issue, if you don't mind

Note : maybe i'm totally wrong on how apps were able, before, to access a specific store, I just deduced from observations

**Comment by JFClaeys** [comment]
ok, tests are working fine, all 3 stores are accessible  after applying your patch.
thanks for your patience and support , appreciated
